zapcore/encoder.go

```diff
@@ -36,18 +55,36 @@ func LowercaseLevelEncoder(l Level, enc PrimitiveArrayEncoder) {
 	enc.AppendString(l.String())
 }
 
+// ColoredLowercaseLevelEncoder serializes a Level to a lowercase string and adds coloring.
+// For example, InfoLevel is serialized to "info".
+func ColoredLowercaseLevelEncoder(l Level, enc PrimitiveArrayEncoder) {
+	enc.AppendString(levelToColor[l].Sprint(l.String()))
```

Comment:
`Sprintf` instead, which saves an allocation. We should also cache the colorized output for the known levels, which should shave off the other allocation.

This also needs to handle map misses; are nil `*color.Color`s safe to use?